File: src/main/java/at/searles/lexer/utils/Counter.java
package at.searles.lexer.utils;

import java.util.Iterator;

public class Counter implements Iterator<Integer> {
    private int i = 0;

    public int incr() {
        return i++;
    }

    public int get() {
        return i;
    }

    @Override
    public boolean hasNext() {
        return true;
    }

    @Override
    public Integer next() {
        return incr();
    }
}

File: src/main/java/at/searles/lexer/utils/Counter.kt
package at.searles.lexer.utils

class Counter: Iterator<Int> {
    private var i = 0
    fun incr(): Int {
        return i++
    }

    fun get(): Int {
        return i
    }

    override fun hasNext(): Boolean {
        return true
    }

    override fun next(): Int {
        return incr()
    }
}